client/components/overviewWidget/AddToCart.jsx

```diff
@@ -39,27 +41,69 @@ class AddToCart extends React.Component {
           console.error(err);
         });
     }
+
+    if (this.props.selectedStyle !== prevProps.selectedStyle) {
+      const newSelectedStyleData = this.state.styles.filter((style) => {
+        return style.style_id === parseInt(this.props.selectedStyle);
+      });
+      let skus = Object.values(newSelectedStyleData[0].skus);
+      let sizes = [];
+      skus.forEach((sku) => {
+        sizes.push(sku.size);
+      });
+      this.setState({
+        selectedStyle: newSelectedStyleData[0],
+        sizes
+      });
+      this.sizeSelectorClickHandler({ target: { value: 'select-size' } });
+    }
+  }
+
+  sizeSelectorClickHandler(e) {
+    const selectedSize = e.target.value;
+    if (selectedSize === 'select-size') {
+      this.setState({
+        selectedSizeQuantities: []
+      });
+    } else {
+      const selectedStyle = this.state.selectedStyle;
+      const selectedSku = Object.values(selectedStyle.skus).filter((sku) => {
+        return sku.size === selectedSize;
+      });
+      const selectedSizeQuantity = selectedSku[0].quantity;
+      const maxAvailableQuantity = Math.min(selectedSizeQuantity, 15);
+      const availableQuantities = [];
+      for (let i = 1; i <= maxAvailableQuantity; i++) {
+        availableQuantities.push(i);
+      }
+      this.setState({
+        selectedSizeQuantities: availableQuantities
+      });
+    }
+  }
```

Comment:
Too long to explain but with a controlled component I think there is a way to make React do this work for you.  If you want to refactor this let me know and I can run my idea by you